### ISSUE TITLE
Server has to be restarted many times a day

### ISSUE THREAD

**Issue by s-h-a-d-o-w** [oxc-project/oxc-vscode#172]
### Confirm that this is a Bug for the editor extension, not the CLI

- [x] This bug does not happen in the Oxlint/Oxfmt CLI
- [ ] After restarting the editor, the bug still appears

### Versions used

```
VS Code Extension v1.50.0
oxfmt is enabled (v0.36.0)
oxlint is enabled (v1.53.0)

VSCode Version: 1.105.1
Commit: b29eb4ee5f9f6d1cb2afbc09070198d3ea6ad760
Date: 2026-03-17T01:50:02.404Z
Build Type: Stable
Release Track: Default
Electron: 39.8.1
Chromium: 142.0.7444.265
Node.js: 22.22.1
V8: 14.2.231.22-electron.0
OS: Windows_NT x64 10.0.26200 - But I'm using WSL! (But it happens in both environments)

Node Manager: nvm
```

### What happened?

Stale state - errors not reported or wrongly reported and I have to restart.
And almost every time I commit, I'll see an error only that didn't appear while I was working on things.

If I'm not the only one this is happening to and you don't know how to diagnose, I would recommend considering routine restart strategies. Like restarting whenever there are changes to the oxlint binary in node_modules (if you don't already). Or whenever the user changes to a different file in the IDE.

### Output channel log

Unfortunately, no errors. Just:
```
2026-03-24 13:04:12.939 [info] 2026-03-24T12:04:12.938796Z  INFO oxc_language_server::backend: initialize: Some([WorkspaceOption { workspace_uri: Uri(Uri { scheme: "file", authority: Some(Authority { userinfo: None, host: "", host_parsed: RegName(""), port: None }), path: "/home/andy/dev/neev", query: None, fragment: None }), options: Object {"typeAware": Bool(true), "disableNestedConfig": Bool(false), "fixKind": String("safe_fix"), "run": String("onType"), "flags": Object {"disable_nested_config": String("false"), "fix_kind": String("safe_fix")}} }])
2026-03-24T12:04:12.938892Z  INFO oxc_language_server::backend: oxlint version: 1.53.0
```

And whenever I restart:
```
2026-03-24 13:04:12.837 [info] 2026-03-24T12:04:12.836777Z  INFO tower_lsp_server::service::layers: shutdown request received, shutting down

2026-03-24 13:04:12.840 [info] 2026-03-24T12:04:12.838521Z  INFO tower_lsp_server::service::layers: exit notification received, stopping

2026-03-24 13:04:12.912 [info] [Error - 1:04:12 PM] Server process exited with code 0.
```

**Comment by Sysix** [comment]
Does this bug only happen when you restarted the linter?
If yes, I hope https://github.com/oxc-project/oxc-vscode/pull/191 will fix it.
